third_party/bigframes_vendored/ibis/backends/sql/compilers/base.py

```diff
@@ -0,0 +1,1660 @@
+# Contains code from https://github.com/ibis-project/ibis/blob/main/ibis/backends/sql/compilers/base.py
+
+from __future__ import annotations
+
+import abc
+import calendar
+from functools import partial, reduce
+import itertools
+import math
+import operator
+import string
+from typing import Any, ClassVar, TYPE_CHECKING
+
+from ibis.backends.sql.rewrites import (
```

Comment:
Should we change this to our vendored copy?

Reply:
Fixed in this PR. Thanks!